### PR TITLE
Freetype: Update to 2.12.0 release

### PR DESCRIPTION
**Pull request by satmandu**
Update freetype to `2.12.0` release.

### CODE REVIEW

**Comment by xclaesse** [comment]
Our policy for wrapdb is to not use wrap-git and only accept wraps based on released tarball.

**Comment by satmandu** [comment]
Is this more acceptable?

**Comment by satmandu** [comment]
(And should a PR update releases.json too?)

File: subprojects/freetype2.wrap [hunk 1]
```diff
@@ -1,8 +1,9 @@
 [wrap-file]
-directory = freetype-2.11.1
-source_url = https://download.savannah.gnu.org/releases/freetype/freetype-2.11.1.tar.gz
-source_filename = freetype-2.11.1.tar.gz
-source_hash = f8db94d307e9c54961b39a1cc799a67d46681480696ed72ecf78d4473770f09b
+directory = freetype
```
Comment (jpakkane) [inline subprojects/freetype2.wrap:2]: This should be the directory that the tar file has, so probably `freetype-2.12.0`.

File: subprojects/freetype2.wrap [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
+source_url = https://download.savannah.gnu.org/releases/freetype/freetype-2.12.0.tar.xz
+source_filename = freetype-2.12.0.tar.xz
+source_hash = ef5c336aacc1a079ff9262d6308d6c2a066dd4d2a905301c4adda9b354399033
 
 [provide]
 freetype2 = freetype_dep
+freetype = freetype_dep
```
Comment (jpakkane) [inline subprojects/freetype2.wrap:9]: FWICT this is not correct. The pkg-config name of Freetype is `freetype2` so this should match that. Just remove this line.

**Review by jpakkane — CHANGES_REQUESTED** [review]

**Comment by jpakkane** [comment]
> (And should a PR update releases.json too?)

Yes.

**Comment by satmandu** [comment]
> > (And should a PR update releases.json too?)
> 
> Yes.

I see releases are suffixed with a "-1" or higher.  So I assume this should be '2.12.0-1' ?

**Comment by xclaesse** [comment]
Exact

**Comment by satmandu** [comment]
done

**Comment by eli-schwartz** [comment]
After the suggested review changes, we are back to the original freetype2.wrap file, but with the version number bumped and the hash updated.

Please squash all changes into one commit. Commit message should probably be something like `Freetype: update to 2.1.12 release`

**Comment by satmandu** [comment]
Adjusted. (Also fixed the title, as we are talking about version 2.12.0...)

**Comment by eli-schwartz** [comment]
I see upstream actually provides both gz and xz, makes sense to switch to the latter indeed.

**Comment by eli-schwartz** [comment]
The macOS CI is erroring out by telling you that this update fixes CI there, so you should remove it from the known-broken list.

**Comment by satmandu** [comment]
Where is that?

**Comment by eli-schwartz** [comment]
The CI message calls out the string `broken_darwin`, you can find this in the file `ci_config.json`.

**Comment by satmandu** [comment]
Done.

**Comment by eli-schwartz** [comment]
Thanks!

**Pull request merged**